src/hotspot/cpu/x86/interp_masm_x86.cpp

```diff
@@ -777,11 +778,56 @@ void InterpreterMacroAssembler::narrow(Register result) {
   bind(done);
 }
 
+void InterpreterMacroAssembler::call_VM_with_sender_Java_fp(Register fp_reg, Register tmp, address entry_point, bool store_bcp) {
+  if (store_bcp) {
+    save_bcp(fp_reg); // We must save the bcp, but need not restore it. This is because we have already popped the fp.
+  }
+  // Adjust sp as we now have a return address on stack.
+  // We've pushed one address, correct last_Java_sp
+  Register last_java_sp = tmp;
+  lea(last_java_sp, Address(rsp, wordSize));
+
+  // Thread argument
+  mov(c_rarg0, r15_thread);
+
+#if INCLUDE_JFR
+  Label L_ljf, L_valid_rbp;
+  testptr(rbp, rbp);
+  jcc(Assembler::notZero, L_valid_rbp);
```

Comment:
Little trivia about x86: "normal" jumps are fairly long (6 bytes, IIRC), but you can have a "short jump" (about 3 bytes, IIRC) if the offset is small. This is normally handled by assembler itself, when it knows where the branch target is. That is known for _backward_ branches. For _forward_ branches like this, we unfortunately often need to claim the jump is short ahead of time. See `jccb` (note `b`). 

Basically, if you are sure the branch target is within 128 bytes (or, more likely, a few instructions) away, use `jccb`.

Same thing with `jmp` -> `jmpb`.